pySDC/implementations/problem_classes/odeSystem.py

```diff
@@ -0,0 +1,925 @@
+
+    def dgInv(self, u, t):
+        raise NotImplementedError()
+
+    def dgInv_LIN(self, u, t, dt):
+        e = self.epsilon
+        g1, g2 = self.dg(t), self.dg2(t)
+        return np.array([[1 / (dt / e + 1), (dt * g2 + dt * g1 / e) / (dt / e + 1)], [0, 1]])
+
+    def dgInv_NONLIN(self, u, t, dt):
+        e = self.epsilon
+        g, g1, g2 = self.g(t), self.dg(t), self.dg2(t)
+        return np.array(
+            [[1 / (3 * dt * u**2 / e + 1), (dt * g2 + 3 * dt * g**2 * g1 / e) / (3 * dt * u**2 / e + 1)], [0, 1]]
+        )
+
+    # -------------------------------------------------------------------------
+    # pySDC required methods
+    # -------------------------------------------------------------------------
+    def u_exact(self, t, u_init=None, t_init=None):
+        r"""
+        Routine to return initial conditions or to approximate exact solution using ``SciPy``.
+
+        Parameters
+        ----------
+        t : float
+            Time at which the approximated exact solution is computed.
+        u_init : pySDC.implementations.problem_classes.Lorenz.dtype_u
+            Initial conditions for getting the exact solution.
+        t_init : float
+            The starting time.
+
+        Returns
+        -------
+        me : dtype_u
+            The approximated exact solution.
+        """
+        u = self.dtype_u(self.init)
+        u[0] = self.g(t)
+        u[1] = t
+        return u
+
+    def eval_f(self, u, t):
+        """
+        Routine to evaluate the right-hand side of the problem.
+
+        Parameters
+        ----------
+        u : dtype_u
+            Current values of the numerical solution.
+        t : float
+            Current time of the numerical solution is computed (not used here).
+
+        Returns
+        -------
+        f : dtype_f
+            The right-hand side of the problem (one component).
+        """
+
+        f = self.dtype_f(self.init)
+        u, t = u
+        f[0] = self.f(u, t)
+        f[1] = 1
+        self.work_counters['rhs']()
+        return f
+
+    def solve_system(self, rhs, dt, u0, t):
+        """
+        Simple Newton solver for the nonlinear equation
+
+        Parameters
+        ----------
+        rhs : dtype_f
+            Right-hand side for the nonlinear system.
+        dt : float
+            Abbrev. for the node-to-node stepsize (or any other factor required).
+        u0 : dtype_u
+            Initial guess for the iterative solver.
+        t : float
+            Time of the updated solution (e.g. for time-dependent BCs).
+
+        Returns
+        -------
+        u : dtype_u
+            The solution as mesh.
+        """
+        # create new mesh object from u0 and set initial values for iteration
+        u = self.dtype_u(u0)
+
+        # start newton iteration
+        n, res = 0, np.inf
+        while n < self.newton_maxiter:
+
+            # form the function g with g(u) = 0
+            g = u - dt * self.eval_f(u, t) - rhs
```

Comment:
If you call the right hand side function in the Newton solver, you increment the associated work counter. Maybe this is desired by you. I prefer to untangle it and decrement the RHS counter in the Newton solver. If you think it makes more sense to add it cost twice in a way, we can do that, but we should do it consistently.

I suspect this comment applies to most problems you implemented. And the behaviour is likely inconsistent with Lorenz and this point. So I deem this important for your paper.

Reply:
Indeed, but this is the only place were it is used. In fact, the RHS is already evaluated before, but not passed to the `solve_system` function. So it make sense not to use the `eval_f` function (as it is done in all the other classes).
This has been corrected now, it only calls the `f` method, which does not increment the RHS counter.

Furthermore,  passing the RHS evaluation in `system_solve` will probably be something to add to the problem rework, see #255

Reply:
Hum, actually I was wrong : the initial evaluation of the RHS can only be used the first time, but then it should be evaluated at each newton iteration. But this does not change the paper results, as this problem class is not used in the numerical experiments.

Also, for most problem class, it does not change the fact that the Newton iteration has almost the same cost as the RHS evaluation ...

Reply:
I think you forgot to switch `eval_f` for `f` in assembling `g`? `f` is not used, I think. You can also do what I did [here](https://github.com/Parallel-in-Time/pySDC/blob/a8a04311a0ce594fed24050c640866809ba6e870/pySDC/implementations/problem_classes/Quench.py#L327), to decrement the work counter for RHS if you prefer.